tensorflow/python/lib/core/ndarray_tensor.cc

```diff
@@ -473,8 +474,20 @@ Status TF_TensorToPyArray(Safe_TF_TensorPtr tensor, PyObject** out_ndarray) {
     if (!s.ok()) {
       return s;
     }
-  } else if (static_cast<size_t>(PyArray_NBYTES(py_array)) !=
-             TF_TensorByteSize(tensor.get())) {
+  } else if ((!IsZenDnnEnabled()) &&
+             (static_cast<size_t>(PyArray_NBYTES(py_array)) !=
+              TF_TensorByteSize(tensor.get()))) {
+    return errors::Internal("ndarray was ", PyArray_NBYTES(py_array),
+                            " bytes but TF_Tensor was ",
+                            TF_TensorByteSize(tensor.get()), " bytes");
+  }
+  // Default TF implementation compares ndarray bytes with TF_Tensor bytes
+  // which is allocated using allocate_output API for fixed size.
+  // ZenDNN mempool optimization utilizes pre-allocated buffers with TF_Tensor.
+  // Hence, amount of allocated TF_Tensor bytes can be greater than or equal
+  // to requested bytes by the Op execution.
```

Comment:
Nit: It wasn't obvious to me that "can be >=" in here means "can't be <" so I think we should make the wording stricter.
```suggestion
  // Hence, the amount of allocated TF_Tensor bytes is always greater than or
  // equal to the number of bytes requested by the op execution.
```

Reply:
Resolved. Thank you for pointing out the issue.